compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -1185,12 +1185,9 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 tcx.sess.struct_span_err(pat.span, "`..` cannot be used in union patterns").emit();
             }
         } else if !etc && !unmentioned_fields.is_empty() {
-            let no_accessible_unmentioned_fields = unmentioned_fields
-                .iter()
-                .find(|(field, _)| {
-                    field.vis.is_accessible_from(tcx.parent_module(pat.hir_id).to_def_id(), tcx)
-                })
-                .is_none();
+            let no_accessible_unmentioned_fields = !unmentioned_fields.iter().any(|(field, _)| {
```

Comment:
Would it make sense for an iterator to have a convenience `none()` method that does the same as `!any()`? We have such methods for `any` and `all`, seems logical to have it for `none` as well.

Reply:
Feel free to open an issue or to propose a PR with an unstable implementation